lib/src/main/java/com/auth0/jwt/impl/PublicClaims.java

```diff
@@ -1,6 +1,8 @@
 package com.auth0.jwt.impl;
 
-
+/**
+ * Contains the claim name for all Public claims.
+ */
 public interface PublicClaims {
```

Comment:
Any reason for this class to be in the `impl` (internal package) ?
I see some other classes using `public` in this package, this is something to consider change in future.
This package should also be removed from module-info (java 9+).

Reply:
My assumption on this class being internal is to provide less API surface area for the consumer. But as you mentioned having it as public negates the effort and we don't have any other option currently since the package structure is that way.

For Java 9+, if the project is using the module feature, then they won't be able to import this package as it can be seen here - https://github.com/auth0/java-jwt/blob/v4-dev/lib/src/main/java/module-info.java.

Reply:
The next major version is a big opportunity to remove public from interfaces/classes in impl package.
Module system is under utilized in Java, even if you using latest Java 18 everyone still relying on classloader.
Lets make sure this get reviewed before release, ideally should have none.

Reply:
I agree @panga, I have created an internal ticket [SDK-3226] to assigned it to this major. Since @jimmyjames has more context over this library, we should decide on this once he is back.